src/util/pkce/__tests__/index.ts

```diff
@@ -0,0 +1,37 @@
+import crypto from 'crypto';
+import { TextEncoder } from 'util';
+import PKCE from '..';
+
+Object.defineProperty(window, 'crypto', {
```

Comment:
Is `crypto` still not available in JSDOM? I noticed you added `jsdom` as an environment for Jest.

Reply:
After seeing that [this Github issue is still open](https://github.com/jsdom/jsdom/issues/1612), I'm assuming `crypto` has still not been implemented, so I see why this was done.